### PR TITLE
PR for Cart Page

### PR DESCRIPTION
**Pull request by kiranpratyush**
Added Cart Page
1. Increment Quantity and price total is changing
2. Handling empty cart with cool animation

https://user-images.githubusercontent.com/77445957/161415795-9426cbea-4513-4eed-860c-0f1b1ce34453.mp4

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,17 +2,19 @@ import React from 'react';
...
   return (
 
     <Routes>
       <Route path ="/" element ={<HomePage/>}></Route>
       <Route path ="/shop" element ={<ShopPage/>}></Route>
-      {/* <Route  path="/cart" element ={<CartPage />}></Route>
-      <Route path ="/wishlist" element ={<WishListPage/>}></Route> */}
+      <Route  path="/cart" element ={<CartPage />}></Route>
+      <Route path ="/wishlist" element ={<WishListPage/>}></Route>
+      <Route path ="/empty" element ={<Empty />}></Route>
```
Comment (Chakravarthi0) [inline src/App.js]: Suggestion: create a separate component for routes

File: src/Components/Card/style.css [hunk 2]
```diff
@@ -15,6 +15,7 @@ img {
   max-width: 460px;
   min-width: 260px;
```
Comment (Chakravarthi0) [inline src/Components/Card/style.css:16]: You can use rem units here

File: src/Contexts/CartContext.js [hunk 3]
```diff
@@ -1,35 +1,90 @@
...
+      const cartData = previousState.data.filter(
+        (element) => element.id === action.payload.id
+      );
+      return {
+        ...previousState,
+        wishList: [...previousState.wishList, wishListdata],
+        data: [...cartData],
+      };
+    case 'MOVE_TO_CART':
+      const cartDataFromWishList = previousState.wishList.filter(
```
Comment (Chakravarthi0) [inline src/Contexts/CartContext.js:32]: Suggestion: create a new file for reducer

**Review by Chakravarthi0 — APPROVED** [review]
Code is clean.
One thing I noticed is that you are not using a backend/making any api calls.
Rest is good.

File: src/App.js [hunk 4]
```diff
@@ -2,17 +2,19 @@ import React from 'react';
 import {Routes,Route} from "react-router-dom"
 import HomePage from "./Pages/HomePage"
 import ShopPage from './Pages/ShopPage';
-// import { CartPage } from './Pages/CartPage';
-// import { WishListPage } from './Pages/WishListPage';
+import { CartPage } from './Pages/CartPage';
+import { WishListPage } from './Pages/WishListPage';
```
Comment (mimansha-swarup) [inline src/App.js:7]: you can use barrel export

File: src/Components/EmptyNotification/Empty.css [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+}
+.empty__container a:hover::after{
+    width:40%;
+    transition: width 0.3s;
+}
+
+@keyframes movein {
+    0% {
+        transform: translateY(-20px);
```
Comment (mimansha-swarup) [inline src/Components/EmptyNotification/Empty.css]: its great that you have added animations

File: src/Contexts/CartContext.js [hunk 6]
```diff
@@ -1,35 +1,90 @@
...
-            return {...previousState}
-    }
-}
 
+import React, { createContext, useContext, useReducer } from 'react';
+
+const CartContext = createContext();
+
+function reducerfn(previousState, action) {
+  switch (action.type) {
```
Comment (mimansha-swarup) [inline src/Contexts/CartContext.js:7]: create separate files for reducer

**Review by mimansha-swarup — APPROVED** [review]

**Review by Shrinkhlarajpoot — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (kiranpratyush) [reply src/App.js]: done

File: src/Contexts/CartContext.js (hunk 3 above)
Comment (kiranpratyush) [reply src/Contexts/CartContext.js:32]: will do

File: src/App.js (hunk 4 above)
Comment (kiranpratyush) [reply src/App.js:7]: will do at last

**Pull request merged**